Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift

```diff
@@ -33,4 +58,94 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
         XCTAssertEqual(currentSitter.bio, newBio)
         XCTAssertEqual(currentSitter.pricePerHour, String(newPricePerHour))
     }
+
+    func testSaveMethodCallsUploadMethodWithoutTriggeringError() async {
+        // Given
+        MockURLProtocol.requestHandler = { request in
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: XCTUnwrap(request.url),
+                    statusCode: 200,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, nil)
+        }
+
+        // When
+        await viewModel.save()
+
+        // Then
+        XCTAssertEqual(viewModel.isErrorOccurred, false)
+    }
+
+    func testSaveMethodHandlesErrorStatusCode() async {
+        // Given
+        MockURLProtocol.requestHandler = { request in
+            let response = try XCTUnwrap(
+                HTTPURLResponse(
+                    url: XCTUnwrap(request.url),
+                    statusCode: 400,
+                    httpVersion: nil,
+                    headerFields: nil
+                )
+            )
+            return (response, nil)
+        }
+
+        // When
+        await viewModel.save()
+
+        // Then
+        XCTAssertEqual(viewModel.isErrorOccurred, true)
+    }
+
+    func testAllErrorStatusCodesInResponseTriggersError() async {
+        // Given
+        for statusCode in 400...499 {
```

Comment:
It's a good idea! 
But maybe it's better to check different error cases than to check one error a hundred times.
```suggestion
        for statusCode in [300, 400, 401, 402, 500] {
```

Reply:
Fixed with e9d20b2